src/js/legend_control.js

```diff
@@ -107,6 +115,9 @@ export default class LegendControl {
     for (let data of this.sections) {
       let section = this.getSection(data);
       if (!section) continue;
+      if (data.id) {
+        section.id = `legend-section-${data.id}`;
```

Comment:
This whole flow is a little hard to read: The `sections` property is assigned by the caller, one of its items has an `id` of `"shields"`, and then 569 lines later we query the document for this computed value.

Anyway, LGTM. Something to think about for a future refactor.

Reply:
The shield section is different from the rest of the legend in numerous ways, to the point that it requires a completely separate code path to generate it in the first place. I suppose it would be worth trying to consolidate the code paths somehow.

One of the differences is that the descriptions get replaced by Wikidata labels asynchronously. Originally, I intended to implement Wikidata labeling functionality for the whole legend, but I reconsidered in favor of eventually maintaining localizations of the static legend entries in this repository: #744. That work will most likely require us to assign an ID to each of the sections and each of their entries anyways.

Previously, all the description rows’ DOM elements, in every section, were obtained from a `querySelectorAll()` call in `open()` then plumbed through to `prettifyNetworkLabels()`. But there’s nothing particularly special about this selector query. Making the element unique in the DOM allows us to query for it at any time. Instead of consolidating the shield section’s code path with the other section generation code, we could split it out into a separate class in a separate file, making it clearer that the shield section is more involved and independent from the rest of the legend.

Reply:
#782 tracks refactoring `LegendControl` to be more digestible.